src/pages/Pairing/Pairing.jsx

```diff
@@ -0,0 +1,109 @@
+import { Button, Container, Grid, Typography } from '@mui/material';
+import { styled } from '@mui/material/styles';
+import React, { useEffect, useState } from 'react';
+import ArrowForwardIcon from '@mui/icons-material/ArrowForward';
+import CheckCircleOutlinedIcon from '@mui/icons-material/CheckCircleOutlined';
+import CourseCard from '../../components/CourseCard/CourseCard';
+
+import styles from './Pairing.module.css';
+import { getCourses } from '../../api/TimetableAPI';
+
+const StyledButton = styled(Button)(() => ({
+  '&.MuiButton-root': {
+    border: '2px #666666 solid',
+  },
+  '&.MuiButton-outlined': {
+    color: '#666666',
+  },
+}));
+
+function Pairing() {
+  const [numOfSelectedCourses, setNumOfSelectedCourses] = useState(0);
+  const [selectedCourses, setSelectedCourses] = useState([]);
+  const [courses, setCourses] = useState([]);
+
+  const handleSelectedCourse = (courseName) => {
+    const numberOfCourse = selectedCourses.length;
+    const filteredSelectedCourse = selectedCourses.filter((value) => courseName !== value);
+
+    if (filteredSelectedCourse.length === numberOfCourse) {
+      setSelectedCourses([...filteredSelectedCourse, courseName]);
+      setNumOfSelectedCourses(numOfSelectedCourses + 1);
```

Comment:
These should probably become `setNumOfSelectedCourses(prev => prev+1)` (and -1 too)